specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/SessionPools.json

```diff
@@ -0,0 +1,702 @@
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-examples": {
+          "Update Session Pool": {
+            "$ref": "./examples/SessionPools_Patch.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "ContainerAppsSessionPools"
+        ],
+        "summary": "Delete a session pool.",
+        "description": "Delete the session pool with the given name.",
+        "operationId": "ContainerAppsSessionPools_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SessionPoolNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Session pool delete has been started.",
+            "headers": {
+              "Location": {
+                "type": "string"
+              }
+            }
+          },
+          "204": {
+            "description": "Session pool does not exist."
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-examples": {
+          "Delete Session Pool": {
+            "$ref": "./examples/SessionPools_Delete.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ScaleConfiguration": {
+      "description": "Scale configuration.",
+      "type": "object",
+      "properties": {
+        "maxConcurrentSessions": {
+          "description": "The maximum count of sessions at the same time.",
+          "type": "integer",
+          "format": "int32"
+        },
+        "readySessionInstances": {
+          "description": "The minimum count of ready session instances.",
+          "type": "integer",
+          "format": "int32"
+        }
+      }
+    },
+    "Secret": {
+      "description": "Secret definition.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Secret Name.",
+          "type": "string"
+        },
+        "value": {
+          "description": "Secret Value.",
+          "type": "string",
+          "x-ms-mutability": [
+            "create",
+            "update"
+          ],
+          "x-ms-secret": true
+        }
+      }
+    },
```

Comment:
I assume you need to model secrets like this because you're providing a general solution for people who are not doing things the cred free way?

Reply:
Yes, and this follows an existing pattern of another API for container app